pandas-stubs/core/frame.pyi

```diff
@@ -114,20 +114,20 @@ class _LocIndexerFrame(_LocIndexer):
         self,
         idx: IndexType
         | MaskType
-        | list[StrLike]
+        | list[HashableT]
         | tuple[
             IndexType
             | MaskType
             | slice
-            | list[StrLike]
+            | list[HashableT]
             | tuple[str | int | slice, ...],
-            list[StrLike] | slice | Series[bool] | Callable,
+            list[HashableT] | slice | Series[bool] | Callable,
         ],
     ) -> DataFrame: ...
     @overload
     def __getitem__(
         self,
-        idx: tuple[int | StrLike | tuple[StrLike, ...], StrLike],
+        idx: tuple[int | StrLike | tuple[HashableT, ...], int | StrLike],
```

Comment:
I think the column indexer should be HashableT too?

Reply:
I tried that, and then `mypy` says the overloads were overlapping (which could be a mypy bug)

Reply:
I assume the first and second overload would overlap then. The first overload accepts `tuple[ tuple[slice, ...], slice]` and the second overload would accept `tuple[tuple[HashableT, ...], HashableT]`. `slice` is `Hashable` and would match both overloads but return different return values.

Reply:
Yes, that must be it.  So I think we have to go with what I put in here

Reply:
Happy to merge after rebasing (conflicts with the other merged PR).

cc @phofl

Reply:
have done the rebasing.  I will do a new release once this is merged.